File: dart/com/spoonacular/client/model/inline_response2003.dart
part of openapi.api;

class InlineResponse2003 {
  
  int id = null;
  
  String title = null;
  
  String image = null;
  
  String imageType = null;
  
  num servings = null;
  
  int readyInMinutes = null;
  
  String license = null;
  
  String sourceName = null;
  
  String sourceUrl = null;
  
  String spoonacularSourceUrl = null;
  
  int aggregateLikes = null;
  
  num healthScore = null;
  
  num spoonacularScore = null;
  
  num pricePerServing = null;
  
  List<Object> analyzedInstructions = [];
  
  bool cheap = null;
  
  String creditsText = null;
  
  List<String> cuisines = [];
  
  bool dairyFree = null;
  
  List<String> diets = [];
  
  String gaps = null;
  
  bool glutenFree = null;
  
  String instructions = null;
  
  bool ketogenic = null;
  
  bool lowFodmap = null;
  
  List<String> occasions = [];
  
  bool sustainable = null;
  
  bool vegan = null;
  
  bool vegetarian = null;
  
  bool veryHealthy = null;
  
  bool veryPopular = null;
  
  bool whole30 = null;
  
  num weightWatcherSmartPoints = null;
  
  List<String> dishTypes = [];
  
  List<InlineResponse2003ExtendedIngredients> extendedIngredients = [];
  
  String summary = null;
  
  InlineResponse2003WinePairing winePairing = null;
  InlineResponse2003();

  @override
  String toString() {
    return 'InlineResponse2003[id=$id, title=$title, image=$image, imageType=$imageType, servings=$servings, readyInMinutes=$readyInMinutes, license=$license, sourceName=$sourceName, sourceUrl=$sourceUrl, spoonacularSourceUrl=$spoonacularSourceUrl, aggregateLikes=$aggregateLikes, healthScore=$healthScore, spoonacularScore=$spoonacularScore, pricePerServing=$pricePerServing, analyzedInstructions=$analyzedInstructions, cheap=$cheap, creditsText=$creditsText, cuisines=$cuisines, dairyFree=$dairyFree, diets=$diets, gaps=$gaps, glutenFree=$glutenFree, instructions=$instructions, ketogenic=$ketogenic, lowFodmap=$lowFodmap, occasions=$occasions, sustainable=$sustainable, vegan=$vegan, vegetarian=$vegetarian, veryHealthy=$veryHealthy, veryPopular=$veryPopular, whole30=$whole30, weightWatcherSmartPoints=$weightWatcherSmartPoints, dishTypes=$dishTypes, extendedIngredients=$extendedIngredients, summary=$summary, winePairing=$winePairing, ]';
  }

  InlineResponse2003.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['id'] == null) {
      id = null;
    } else {
          id = json['id'];
    }
    if (json['title'] == null) {
      title = null;
    } else {
          title = json['title'];
    }
    if (json['image'] == null) {
      image = null;
    } else {
          image = json['image'];
    }
    if (json['imageType'] == null) {
      imageType = null;
    } else {
          imageType = json['imageType'];
    }
    if (json['servings'] == null) {
      servings = null;
    } else {
          servings = json['servings'];
    }
    if (json['readyInMinutes'] == null) {
      readyInMinutes = null;
    } else {
          readyInMinutes = json['readyInMinutes'];
    }
    if (json['license'] == null) {
      license = null;
    } else {
          license = json['license'];
    }
    if (json['sourceName'] == null) {
      sourceName = null;
    } else {
          sourceName = json['sourceName'];
    }
    if (json['sourceUrl'] == null) {
      sourceUrl = null;
    } else {
          sourceUrl = json['sourceUrl'];
    }
    if (json['spoonacularSourceUrl'] == null) {
      spoonacularSourceUrl = null;
    } else {
          spoonacularSourceUrl = json['spoonacularSourceUrl'];
    }
    if (json['aggregateLikes'] == null) {
      aggregateLikes = null;
    } else {
          aggregateLikes = json['aggregateLikes'];
    }
    if (json['healthScore'] == null) {
      healthScore = null;
    } else {
          healthScore = json['healthScore'];
    }
    if (json['spoonacularScore'] == null) {
      spoonacularScore = null;
    } else {
          spoonacularScore = json['spoonacularScore'];
    }
    if (json['pricePerServing'] == null) {
      pricePerServing = null;
    } else {
          pricePerServing = json['pricePerServing'];
    }
    if (json['analyzedInstructions'] == null) {
      analyzedInstructions = null;
    } else {
      analyzedInstructions = Object.listFromJson(json['analyzedInstructions']);
    }
    if (json['cheap'] == null) {
      cheap = null;
    } else {
          cheap = json['cheap'];
    }
    if (json['creditsText'] == null) {
      creditsText = null;
    } else {
          creditsText = json['creditsText'];
    }
    if (json['cuisines'] == null) {
      cuisines = null;
    } else {
      cuisines = (json['cuisines'] as List).cast<String>();
    }
    if (json['dairyFree'] == null) {
      dairyFree = null;
    } else {
          dairyFree = json['dairyFree'];
    }
    if (json['diets'] == null) {
      diets = null;
    } else {
      diets = (json['diets'] as List).cast<String>();
    }
    if (json['gaps'] == null) {
      gaps = null;
    } else {
          gaps = json['gaps'];
    }
    if (json['glutenFree'] == null) {
      glutenFree = null;
    } else {
          glutenFree = json['glutenFree'];
    }
    if (json['instructions'] == null) {
      instructions = null;
    } else {
          instructions = json['instructions'];
    }
    if (json['ketogenic'] == null) {
      ketogenic = null;
    } else {
          ketogenic = json['ketogenic'];
    }
    if (json['lowFodmap'] == null) {
      lowFodmap = null;
    } else {
          lowFodmap = json['lowFodmap'];
    }
    if (json['occasions'] == null) {
      occasions = null;
    } else {
      occasions = (json['occasions'] as List).cast<String>();
    }
    if (json['sustainable'] == null) {
      sustainable = null;
    } else {
          sustainable = json['sustainable'];
    }
    if (json['vegan'] == null) {
      vegan = null;
    } else {
          vegan = json['vegan'];
    }
    if (json['vegetarian'] == null) {
      vegetarian = null;
    } else {
          vegetarian = json['vegetarian'];
    }
    if (json['veryHealthy'] == null) {
      veryHealthy = null;
    } else {
          veryHealthy = json['veryHealthy'];
    }
    if (json['veryPopular'] == null) {
      veryPopular = null;
    } else {
          veryPopular = json['veryPopular'];
    }
    if (json['whole30'] == null) {
      whole30 = null;
    } else {
          whole30 = json['whole30'];
    }
    if (json['weightWatcherSmartPoints'] == null) {
      weightWatcherSmartPoints = null;
    } else {
          weightWatcherSmartPoints = json['weightWatcherSmartPoints'];
    }
    if (json['dishTypes'] == null) {
      dishTypes = null;
    } else {
      dishTypes = (json['dishTypes'] as List).cast<String>();
    }
    if (json['extendedIngredients'] == null) {
      extendedIngredients = null;
    } else {
      extendedIngredients = InlineResponse2003ExtendedIngredients.listFromJson(json['extendedIngredients']);
    }
    if (json['summary'] == null) {
      summary = null;
    } else {
          summary = json['summary'];
    }
    if (json['winePairing'] == null) {
      winePairing = null;
    } else {
      winePairing = new InlineResponse2003WinePairing.fromJson(json['winePairing']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (id != null)
      json['id'] = id;
    if (title != null)
      json['title'] = title;
    if (image != null)
      json['image'] = image;
    if (imageType != null)
      json['imageType'] = imageType;
    if (servings != null)
      json['servings'] = servings;
    if (readyInMinutes != null)
      json['readyInMinutes'] = readyInMinutes;
    if (license != null)
      json['license'] = license;
    if (sourceName != null)
      json['sourceName'] = sourceName;
    if (sourceUrl != null)
      json['sourceUrl'] = sourceUrl;
    if (spoonacularSourceUrl != null)
      json['spoonacularSourceUrl'] = spoonacularSourceUrl;
    if (aggregateLikes != null)
      json['aggregateLikes'] = aggregateLikes;
    if (healthScore != null)
      json['healthScore'] = healthScore;
    if (spoonacularScore != null)
      json['spoonacularScore'] = spoonacularScore;
    if (pricePerServing != null)
      json['pricePerServing'] = pricePerServing;
    if (analyzedInstructions != null)
      json['analyzedInstructions'] = analyzedInstructions;
    if (cheap != null)
      json['cheap'] = cheap;
    if (creditsText != null)
      json['creditsText'] = creditsText;
    if (cuisines != null)
      json['cuisines'] = cuisines;
    if (dairyFree != null)
      json['dairyFree'] = dairyFree;
    if (diets != null)
      json['diets'] = diets;
    if (gaps != null)
      json['gaps'] = gaps;
    if (glutenFree != null)
      json['glutenFree'] = glutenFree;
    if (instructions != null)
      json['instructions'] = instructions;
    if (ketogenic != null)
      json['ketogenic'] = ketogenic;
    if (lowFodmap != null)
      json['lowFodmap'] = lowFodmap;
    if (occasions != null)
      json['occasions'] = occasions;
    if (sustainable != null)
      json['sustainable'] = sustainable;
    if (vegan != null)
      json['vegan'] = vegan;
    if (vegetarian != null)
      json['vegetarian'] = vegetarian;
    if (veryHealthy != null)
      json['veryHealthy'] = veryHealthy;
    if (veryPopular != null)
      json['veryPopular'] = veryPopular;
    if (whole30 != null)
      json['whole30'] = whole30;
    if (weightWatcherSmartPoints != null)
      json['weightWatcherSmartPoints'] = weightWatcherSmartPoints;
    if (dishTypes != null)
      json['dishTypes'] = dishTypes;
    if (extendedIngredients != null)
      json['extendedIngredients'] = extendedIngredients;
    if (summary != null)
      json['summary'] = summary;
    if (winePairing != null)
      json['winePairing'] = winePairing;
    return json;
  }

  static List<InlineResponse2003> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse2003>() : json.map((value) => new InlineResponse2003.fromJson(value)).toList();
  }

  static Map<String, InlineResponse2003> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse2003>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse2003.fromJson(value));
    }
    return map;
  }
}

File: android/src/main/java/com/spoonacular/client/model/InlineResponse2003.java
package com.spoonacular.client.model;

import com.spoonacular.client.model.InlineResponse2003ExtendedIngredients;
import com.spoonacular.client.model.InlineResponse2003WinePairing;
import java.math.BigDecimal;
import java.util.*;

import io.swagger.annotations.*;
import com.google.gson.annotations.SerializedName;


@ApiModel(description = "")
public class InlineResponse2003  {
  
  @SerializedName("id")
  private Integer id = null;
  @SerializedName("title")
  private String title = null;
  @SerializedName("image")
  private String image = null;
  @SerializedName("imageType")
  private String imageType = null;
  @SerializedName("servings")
  private BigDecimal servings = null;
  @SerializedName("readyInMinutes")
  private Integer readyInMinutes = null;
  @SerializedName("license")
  private String license = null;
  @SerializedName("sourceName")
  private String sourceName = null;
  @SerializedName("sourceUrl")
  private String sourceUrl = null;
  @SerializedName("spoonacularSourceUrl")
  private String spoonacularSourceUrl = null;
  @SerializedName("aggregateLikes")
  private Integer aggregateLikes = null;
  @SerializedName("healthScore")
  private BigDecimal healthScore = null;
  @SerializedName("spoonacularScore")
  private BigDecimal spoonacularScore = null;
  @SerializedName("pricePerServing")
  private BigDecimal pricePerServing = null;
  @SerializedName("analyzedInstructions")
  private List<Object> analyzedInstructions = null;
  @SerializedName("cheap")
  private Boolean cheap = null;
  @SerializedName("creditsText")
  private String creditsText = null;
  @SerializedName("cuisines")
  private List<String> cuisines = null;
  @SerializedName("dairyFree")
  private Boolean dairyFree = null;
  @SerializedName("diets")
  private List<String> diets = null;
  @SerializedName("gaps")
  private String gaps = null;
  @SerializedName("glutenFree")
  private Boolean glutenFree = null;
  @SerializedName("instructions")
  private String instructions = null;
  @SerializedName("ketogenic")
  private Boolean ketogenic = null;
  @SerializedName("lowFodmap")
  private Boolean lowFodmap = null;
  @SerializedName("occasions")
  private List<String> occasions = null;
  @SerializedName("sustainable")
  private Boolean sustainable = null;
  @SerializedName("vegan")
  private Boolean vegan = null;
  @SerializedName("vegetarian")
  private Boolean vegetarian = null;
  @SerializedName("veryHealthy")
  private Boolean veryHealthy = null;
  @SerializedName("veryPopular")
  private Boolean veryPopular = null;
  @SerializedName("whole30")
  private Boolean whole30 = null;
  @SerializedName("weightWatcherSmartPoints")
  private BigDecimal weightWatcherSmartPoints = null;
  @SerializedName("dishTypes")
  private List<String> dishTypes = null;
  @SerializedName("extendedIngredients")
  private List<InlineResponse2003ExtendedIngredients> extendedIngredients = null;
  @SerializedName("summary")
  private String summary = null;
  @SerializedName("winePairing")
  private InlineResponse2003WinePairing winePairing = null;

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Integer getId() {
    return id;
  }
  public void setId(Integer id) {
    this.id = id;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getTitle() {
    return title;
  }
  public void setTitle(String title) {
    this.title = title;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getImage() {
    return image;
  }
  public void setImage(String image) {
    this.image = image;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getImageType() {
    return imageType;
  }
  public void setImageType(String imageType) {
    this.imageType = imageType;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public BigDecimal getServings() {
    return servings;
  }
  public void setServings(BigDecimal servings) {
    this.servings = servings;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Integer getReadyInMinutes() {
    return readyInMinutes;
  }
  public void setReadyInMinutes(Integer readyInMinutes) {
    this.readyInMinutes = readyInMinutes;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getLicense() {
    return license;
  }
  public void setLicense(String license) {
    this.license = license;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getSourceName() {
    return sourceName;
  }
  public void setSourceName(String sourceName) {
    this.sourceName = sourceName;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getSourceUrl() {
    return sourceUrl;
  }
  public void setSourceUrl(String sourceUrl) {
    this.sourceUrl = sourceUrl;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getSpoonacularSourceUrl() {
    return spoonacularSourceUrl;
  }
  public void setSpoonacularSourceUrl(String spoonacularSourceUrl) {
    this.spoonacularSourceUrl = spoonacularSourceUrl;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Integer getAggregateLikes() {
    return aggregateLikes;
  }
  public void setAggregateLikes(Integer aggregateLikes) {
    this.aggregateLikes = aggregateLikes;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public BigDecimal getHealthScore() {
    return healthScore;
  }
  public void setHealthScore(BigDecimal healthScore) {
    this.healthScore = healthScore;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public BigDecimal getSpoonacularScore() {
    return spoonacularScore;
  }
  public void setSpoonacularScore(BigDecimal spoonacularScore) {
    this.spoonacularScore = spoonacularScore;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public BigDecimal getPricePerServing() {
    return pricePerServing;
  }
  public void setPricePerServing(BigDecimal pricePerServing) {
    this.pricePerServing = pricePerServing;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public List<Object> getAnalyzedInstructions() {
    return analyzedInstructions;
  }
  public void setAnalyzedInstructions(List<Object> analyzedInstructions) {
    this.analyzedInstructions = analyzedInstructions;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getCheap() {
    return cheap;
  }
  public void setCheap(Boolean cheap) {
    this.cheap = cheap;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getCreditsText() {
    return creditsText;
  }
  public void setCreditsText(String creditsText) {
    this.creditsText = creditsText;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public List<String> getCuisines() {
    return cuisines;
  }
  public void setCuisines(List<String> cuisines) {
    this.cuisines = cuisines;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getDairyFree() {
    return dairyFree;
  }
  public void setDairyFree(Boolean dairyFree) {
    this.dairyFree = dairyFree;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public List<String> getDiets() {
    return diets;
  }
  public void setDiets(List<String> diets) {
    this.diets = diets;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getGaps() {
    return gaps;
  }
  public void setGaps(String gaps) {
    this.gaps = gaps;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getGlutenFree() {
    return glutenFree;
  }
  public void setGlutenFree(Boolean glutenFree) {
    this.glutenFree = glutenFree;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getInstructions() {
    return instructions;
  }
  public void setInstructions(String instructions) {
    this.instructions = instructions;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getKetogenic() {
    return ketogenic;
  }
  public void setKetogenic(Boolean ketogenic) {
    this.ketogenic = ketogenic;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getLowFodmap() {
    return lowFodmap;
  }
  public void setLowFodmap(Boolean lowFodmap) {
    this.lowFodmap = lowFodmap;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public List<String> getOccasions() {
    return occasions;
  }
  public void setOccasions(List<String> occasions) {
    this.occasions = occasions;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getSustainable() {
    return sustainable;
  }
  public void setSustainable(Boolean sustainable) {
    this.sustainable = sustainable;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getVegan() {
    return vegan;
  }
  public void setVegan(Boolean vegan) {
    this.vegan = vegan;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getVegetarian() {
    return vegetarian;
  }
  public void setVegetarian(Boolean vegetarian) {
    this.vegetarian = vegetarian;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getVeryHealthy() {
    return veryHealthy;
  }
  public void setVeryHealthy(Boolean veryHealthy) {
    this.veryHealthy = veryHealthy;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getVeryPopular() {
    return veryPopular;
  }
  public void setVeryPopular(Boolean veryPopular) {
    this.veryPopular = veryPopular;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public Boolean getWhole30() {
    return whole30;
  }
  public void setWhole30(Boolean whole30) {
    this.whole30 = whole30;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public BigDecimal getWeightWatcherSmartPoints() {
    return weightWatcherSmartPoints;
  }
  public void setWeightWatcherSmartPoints(BigDecimal weightWatcherSmartPoints) {
    this.weightWatcherSmartPoints = weightWatcherSmartPoints;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public List<String> getDishTypes() {
    return dishTypes;
  }
  public void setDishTypes(List<String> dishTypes) {
    this.dishTypes = dishTypes;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public List<InlineResponse2003ExtendedIngredients> getExtendedIngredients() {
    return extendedIngredients;
  }
  public void setExtendedIngredients(List<InlineResponse2003ExtendedIngredients> extendedIngredients) {
    this.extendedIngredients = extendedIngredients;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public String getSummary() {
    return summary;
  }
  public void setSummary(String summary) {
    this.summary = summary;
  }

  /**
   **/
  @ApiModelProperty(required = true, value = "")
  public InlineResponse2003WinePairing getWinePairing() {
    return winePairing;
  }
  public void setWinePairing(InlineResponse2003WinePairing winePairing) {
    this.winePairing = winePairing;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    InlineResponse2003 inlineResponse2003 = (InlineResponse2003) o;
    return (this.id == null ? inlineResponse2003.id == null : this.id.equals(inlineResponse2003.id)) &&
        (this.title == null ? inlineResponse2003.title == null : this.title.equals(inlineResponse2003.title)) &&
        (this.image == null ? inlineResponse2003.image == null : this.image.equals(inlineResponse2003.image)) &&
        (this.imageType == null ? inlineResponse2003.imageType == null : this.imageType.equals(inlineResponse2003.imageType)) &&
        (this.servings == null ? inlineResponse2003.servings == null : this.servings.equals(inlineResponse2003.servings)) &&
        (this.readyInMinutes == null ? inlineResponse2003.readyInMinutes == null : this.readyInMinutes.equals(inlineResponse2003.readyInMinutes)) &&
        (this.license == null ? inlineResponse2003.license == null : this.license.equals(inlineResponse2003.license)) &&
        (this.sourceName == null ? inlineResponse2003.sourceName == null : this.sourceName.equals(inlineResponse2003.sourceName)) &&
        (this.sourceUrl == null ? inlineResponse2003.sourceUrl == null : this.sourceUrl.equals(inlineResponse2003.sourceUrl)) &&
        (this.spoonacularSourceUrl == null ? inlineResponse2003.spoonacularSourceUrl == null : this.spoonacularSourceUrl.equals(inlineResponse2003.spoonacularSourceUrl)) &&
        (this.aggregateLikes == null ? inlineResponse2003.aggregateLikes == null : this.aggregateLikes.equals(inlineResponse2003.aggregateLikes)) &&
        (this.healthScore == null ? inlineResponse2003.healthScore == null : this.healthScore.equals(inlineResponse2003.healthScore)) &&
        (this.spoonacularScore == null ? inlineResponse2003.spoonacularScore == null : this.spoonacularScore.equals(inlineResponse2003.spoonacularScore)) &&
        (this.pricePerServing == null ? inlineResponse2003.pricePerServing == null : this.pricePerServing.equals(inlineResponse2003.pricePerServing)) &&
        (this.analyzedInstructions == null ? inlineResponse2003.analyzedInstructions == null : this.analyzedInstructions.equals(inlineResponse2003.analyzedInstructions)) &&
        (this.cheap == null ? inlineResponse2003.cheap == null : this.cheap.equals(inlineResponse2003.cheap)) &&
        (this.creditsText == null ? inlineResponse2003.creditsText == null : this.creditsText.equals(inlineResponse2003.creditsText)) &&
        (this.cuisines == null ? inlineResponse2003.cuisines == null : this.cuisines.equals(inlineResponse2003.cuisines)) &&
        (this.dairyFree == null ? inlineResponse2003.dairyFree == null : this.dairyFree.equals(inlineResponse2003.dairyFree)) &&
        (this.diets == null ? inlineResponse2003.diets == null : this.diets.equals(inlineResponse2003.diets)) &&
        (this.gaps == null ? inlineResponse2003.gaps == null : this.gaps.equals(inlineResponse2003.gaps)) &&
        (this.glutenFree == null ? inlineResponse2003.glutenFree == null : this.glutenFree.equals(inlineResponse2003.glutenFree)) &&
        (this.instructions == null ? inlineResponse2003.instructions == null : this.instructions.equals(inlineResponse2003.instructions)) &&
        (this.ketogenic == null ? inlineResponse2003.ketogenic == null : this.ketogenic.equals(inlineResponse2003.ketogenic)) &&
        (this.lowFodmap == null ? inlineResponse2003.lowFodmap == null : this.lowFodmap.equals(inlineResponse2003.lowFodmap)) &&
        (this.occasions == null ? inlineResponse2003.occasions == null : this.occasions.equals(inlineResponse2003.occasions)) &&
        (this.sustainable == null ? inlineResponse2003.sustainable == null : this.sustainable.equals(inlineResponse2003.sustainable)) &&
        (this.vegan == null ? inlineResponse2003.vegan == null : this.vegan.equals(inlineResponse2003.vegan)) &&
        (this.vegetarian == null ? inlineResponse2003.vegetarian == null : this.vegetarian.equals(inlineResponse2003.vegetarian)) &&
        (this.veryHealthy == null ? inlineResponse2003.veryHealthy == null : this.veryHealthy.equals(inlineResponse2003.veryHealthy)) &&
        (this.veryPopular == null ? inlineResponse2003.veryPopular == null : this.veryPopular.equals(inlineResponse2003.veryPopular)) &&
        (this.whole30 == null ? inlineResponse2003.whole30 == null : this.whole30.equals(inlineResponse2003.whole30)) &&
        (this.weightWatcherSmartPoints == null ? inlineResponse2003.weightWatcherSmartPoints == null : this.weightWatcherSmartPoints.equals(inlineResponse2003.weightWatcherSmartPoints)) &&
        (this.dishTypes == null ? inlineResponse2003.dishTypes == null : this.dishTypes.equals(inlineResponse2003.dishTypes)) &&
        (this.extendedIngredients == null ? inlineResponse2003.extendedIngredients == null : this.extendedIngredients.equals(inlineResponse2003.extendedIngredients)) &&
        (this.summary == null ? inlineResponse2003.summary == null : this.summary.equals(inlineResponse2003.summary)) &&
        (this.winePairing == null ? inlineResponse2003.winePairing == null : this.winePairing.equals(inlineResponse2003.winePairing));
  }

  @Override
  public int hashCode() {
    int result = 17;
    result = 31 * result + (this.id == null ? 0: this.id.hashCode());
    result = 31 * result + (this.title == null ? 0: this.title.hashCode());
    result = 31 * result + (this.image == null ? 0: this.image.hashCode());
    result = 31 * result + (this.imageType == null ? 0: this.imageType.hashCode());
    result = 31 * result + (this.servings == null ? 0: this.servings.hashCode());
    result = 31 * result + (this.readyInMinutes == null ? 0: this.readyInMinutes.hashCode());
    result = 31 * result + (this.license == null ? 0: this.license.hashCode());
    result = 31 * result + (this.sourceName == null ? 0: this.sourceName.hashCode());
    result = 31 * result + (this.sourceUrl == null ? 0: this.sourceUrl.hashCode());
    result = 31 * result + (this.spoonacularSourceUrl == null ? 0: this.spoonacularSourceUrl.hashCode());
    result = 31 * result + (this.aggregateLikes == null ? 0: this.aggregateLikes.hashCode());
    result = 31 * result + (this.healthScore == null ? 0: this.healthScore.hashCode());
    result = 31 * result + (this.spoonacularScore == null ? 0: this.spoonacularScore.hashCode());
    result = 31 * result + (this.pricePerServing == null ? 0: this.pricePerServing.hashCode());
    result = 31 * result + (this.analyzedInstructions == null ? 0: this.analyzedInstructions.hashCode());
    result = 31 * result + (this.cheap == null ? 0: this.cheap.hashCode());
    result = 31 * result + (this.creditsText == null ? 0: this.creditsText.hashCode());
    result = 31 * result + (this.cuisines == null ? 0: this.cuisines.hashCode());
    result = 31 * result + (this.dairyFree == null ? 0: this.dairyFree.hashCode());
    result = 31 * result + (this.diets == null ? 0: this.diets.hashCode());
    result = 31 * result + (this.gaps == null ? 0: this.gaps.hashCode());
    result = 31 * result + (this.glutenFree == null ? 0: this.glutenFree.hashCode());
    result = 31 * result + (this.instructions == null ? 0: this.instructions.hashCode());
    result = 31 * result + (this.ketogenic == null ? 0: this.ketogenic.hashCode());
    result = 31 * result + (this.lowFodmap == null ? 0: this.lowFodmap.hashCode());
    result = 31 * result + (this.occasions == null ? 0: this.occasions.hashCode());
    result = 31 * result + (this.sustainable == null ? 0: this.sustainable.hashCode());
    result = 31 * result + (this.vegan == null ? 0: this.vegan.hashCode());
    result = 31 * result + (this.vegetarian == null ? 0: this.vegetarian.hashCode());
    result = 31 * result + (this.veryHealthy == null ? 0: this.veryHealthy.hashCode());
    result = 31 * result + (this.veryPopular == null ? 0: this.veryPopular.hashCode());
    result = 31 * result + (this.whole30 == null ? 0: this.whole30.hashCode());
    result = 31 * result + (this.weightWatcherSmartPoints == null ? 0: this.weightWatcherSmartPoints.hashCode());
    result = 31 * result + (this.dishTypes == null ? 0: this.dishTypes.hashCode());
    result = 31 * result + (this.extendedIngredients == null ? 0: this.extendedIngredients.hashCode());
    result = 31 * result + (this.summary == null ? 0: this.summary.hashCode());
    result = 31 * result + (this.winePairing == null ? 0: this.winePairing.hashCode());
    return result;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class InlineResponse2003 {\n");
    
    sb.append("  id: ").append(id).append("\n");
    sb.append("  title: ").append(title).append("\n");
    sb.append("  image: ").append(image).append("\n");
    sb.append("  imageType: ").append(imageType).append("\n");
    sb.append("  servings: ").append(servings).append("\n");
    sb.append("  readyInMinutes: ").append(readyInMinutes).append("\n");
    sb.append("  license: ").append(license).append("\n");
    sb.append("  sourceName: ").append(sourceName).append("\n");
    sb.append("  sourceUrl: ").append(sourceUrl).append("\n");
    sb.append("  spoonacularSourceUrl: ").append(spoonacularSourceUrl).append("\n");
    sb.append("  aggregateLikes: ").append(aggregateLikes).append("\n");
    sb.append("  healthScore: ").append(healthScore).append("\n");
    sb.append("  spoonacularScore: ").append(spoonacularScore).append("\n");
    sb.append("  pricePerServing: ").append(pricePerServing).append("\n");
    sb.append("  analyzedInstructions: ").append(analyzedInstructions).append("\n");
    sb.append("  cheap: ").append(cheap).append("\n");
    sb.append("  creditsText: ").append(creditsText).append("\n");
    sb.append("  cuisines: ").append(cuisines).append("\n");
    sb.append("  dairyFree: ").append(dairyFree).append("\n");
    sb.append("  diets: ").append(diets).append("\n");
    sb.append("  gaps: ").append(gaps).append("\n");
    sb.append("  glutenFree: ").append(glutenFree).append("\n");
    sb.append("  instructions: ").append(instructions).append("\n");
    sb.append("  ketogenic: ").append(ketogenic).append("\n");
    sb.append("  lowFodmap: ").append(lowFodmap).append("\n");
    sb.append("  occasions: ").append(occasions).append("\n");
    sb.append("  sustainable: ").append(sustainable).append("\n");
    sb.append("  vegan: ").append(vegan).append("\n");
    sb.append("  vegetarian: ").append(vegetarian).append("\n");
    sb.append("  veryHealthy: ").append(veryHealthy).append("\n");
    sb.append("  veryPopular: ").append(veryPopular).append("\n");
    sb.append("  whole30: ").append(whole30).append("\n");
    sb.append("  weightWatcherSmartPoints: ").append(weightWatcherSmartPoints).append("\n");
    sb.append("  dishTypes: ").append(dishTypes).append("\n");
    sb.append("  extendedIngredients: ").append(extendedIngredients).append("\n");
    sb.append("  summary: ").append(summary).append("\n");
    sb.append("  winePairing: ").append(winePairing).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}
